tests/compiler/std/map.optimized.wat

```diff
@@ -2559,10 +2546,9 @@
       i32.const 0
      else
       local.get $1
-      i32.load8_u
+      i32.load8_s
       local.get $3
-      i32.const 255
-      i32.and
+      i32.extend8_s
```

Comment:
That's looks strange little bit. Is it correct changing `load8_u` to `load8_s` and zero-extension to sign-extension?

Reply:
Hmm, good question. Looks like it's picking the wrong sign extension instruction indeed.

Reply:
Every unsigned load and every zero-extensions was changed to signed versions. I don't think it's correct. Is it doing by binaryen, right?

Reply:
Might be a Binaryen optimization doing this, at least didn't find it in the untouched file on a first glimpse. Also, the only place we are emitting `i32.extend8_s` is in `ensureSmallIntegerWrap` on `Type.i8`, which seems correct.

Reply:
zero-extensions should leave as is. I don't understand why it forced always to sign-extensions

Reply:
However it doing only for cases like `(x & 0xFFFF) == y` --> `i32.extrend16_s(x) == y` which seems legit due to `i32.eq` / `i32.ne` is sign-agnistic

Reply:
Just to confirm: The optimization here is actually correct, so nothing to worry about?

Reply:
yes, it's correct